pages/catalog/CatalogHandler.inc.php

```diff
@@ -51,21 +51,91 @@ function authorize($request, &$args, $roleAssignments) {
 	 * @param $request PKPRequest
 	 */
 	function index($args, $request) {
+		$this->page($args, $request, true);
+	}
+
+	/**
+	 * Show one page from the catalog
+	 * @param $args array
+	 * @param $request Request
+	 * @param $isFirstPage boolean
+	 */
+	public function page($args, $request, $isFirstPage) {
+		$page = null;
+		if ($isFirstPage) {
+			$page = 1;
+		} elseif ($args[0]) {
+			$page = (int) $args[0];
+		}
+
+		if (!$isFirstPage && (empty($page) || $page < 2)) {
+			$request->getDispatcher()->handle404();
+		}
+
 		$templateMgr = TemplateManager::getManager($request);
 		$this->setupTemplate($request);
 		$press = $request->getPress();
 
 		// Fetch the monographs to display
+		$itemsPerPage = $press->getData('itemsPerPage');
+		$rangeInfo = new DBResultRange($itemsPerPage, $page);
 		$publishedMonographDao = DAORegistry::getDAO('PublishedMonographDAO');
-		$publishedMonographs = $publishedMonographDao->getByPressId($press->getId());
-		$templateMgr->assign('publishedMonographs', $publishedMonographs->toAssociativeArray());
+		$publishedMonographs = $publishedMonographDao->getByPressId($press->getId(), null, $rangeInfo, null, null, null, null, true);
 
-		// Expose the featured monograph IDs and associated params
-		$featureDao = DAORegistry::getDAO('FeatureDAO');
-		$featuredMonographIds = $featureDao->getSequencesByAssoc(ASSOC_TYPE_PRESS, $press->getId());
-		$templateMgr->assign('featuredMonographIds', $featuredMonographIds);
+		// The DAOResultFactory will return the last available page of results. If
+		// it's less than the requested page, we should not display any results.
+		if ($publishedMonographs->page < $page) {
```

Comment:
Better to use `getPageCount` (as below) rather than accessing `page` directly -- ideally `page` should be declared protected.